### PR TITLE
Add vectorized signatures for get_real and get_imag

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Run unit tests
- Documentation
    - [x] If a user-facing facing change was made, the documentation PR is here: https://github.com/stan-dev/docs/pull/515

## Release notes

Added vectorized signatures for `get_real` and `get_imag`, e.g. `get_real(complex_matrix) => matrix`

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: src/middle/Stan_math_signatures.ml [hunk 1]
```diff
@@ -103,6 +103,14 @@ let rec ints_to_real unsized =
   | UArray t -> UArray (ints_to_real t)
   | x -> x
 
+let rec complex_to_real = function
```
Comment (SteveBronder) [inline src/middle/Stan_math_signatures.ml:106]: Can we put this in UnsizedType?

**Review by SteveBronder — CHANGES_REQUESTED** [review]
One small change then this is good

File: src/middle/Stan_math_signatures.ml (hunk 1 above)
Comment (WardBrian) [reply src/middle/Stan_math_signatures.ml:106]: We could but I don't think we should. Like the function above it (`ints_to_real`) I think the only safe uses for it in the compiler are in writing signatures. If you can think of other places we would want it I can move it over

**Review by SteveBronder — APPROVED** [review]
Good!

**Pull request merged**